### PR TITLE
Support implicit decorations from the Metals server

### PR DESCRIPTION
**Pull request by tgodzik**
This only influences decorations that are inserted inside text. As I checked it does not break worksheets.

Depends on https://github.com/scalameta/metals-languageclient/pull/188

### CODE REVIEW

**Review by gabro — APPROVED** [review]
👍

**Comment by kpbochenek** [comment]
Should we expose those 2 configuration parameters to the user?

**Comment by tgodzik** [comment]
> Should we expose those 2 configuration parameters to the user?

Yeah, it was there at some point, but disappeared :/

**Comment by tgodzik** [comment]
> Should we expose those 2 configuration parameters to the user?

And fixed! Thanks for noticing!

File: package.json [hunk 1]
```diff
@@ -123,6 +123,14 @@
           "default": [],
           "markdownDescription": "List of packages you'd like to be left out of completions, symbol searches, and code actions.\n\nEx. `akka.actor.typed.javadsl` will ensure nothing in the `javadsl` package gets recommended to you.\n\nYou can find the list of default exclusions [here on the Metals website](https://scalameta.org/metals/docs/editors/new-editor.html#excluded-packages).\n\nIf you need to remove one of the defaults, you can simply include it and preface it with `--`."
         },
+        "metals.typeAnnotationsEnabled": {
+          "type": "boolean",
+          "markdownDescription": "When this option is enabled, for each method that can have infered types they are displayed either as additional decorations if they are supported for the editor or as info diagnostics otherwise."
```
Comment (kpbochenek) [inline package.json:128]: Okay so typo 'inferred'
But we can also be more specific, sth like:
`for each method that can have inferred types they are displayed as additional decorations.`

File: package.json [hunk 2]
```diff
@@ -123,6 +123,14 @@
...
+        },
```
Comment (kpbochenek) [inline package.json:129]: And a question to 'can have inferred types', when will method have type parameter but it won't be displayed when this is enabled? Ignoring dotty :rofl:

**Pull request merged**